server/app/models/stack_deploy.rb

```diff
@@ -1,12 +1,66 @@
 class StackDeploy
   include Mongoid::Document
   include Mongoid::Timestamps
-  include Mongoid::Enum
 
-  index({ stack_id: 1 })
+  field :error, type: String
+  field :error_at, type: DateTime
 
-  enum :deploy_state, [:created, :ongoing, :success, :error], default: :created
+  index({ stack_id: 1 })
 
   belongs_to :stack
   has_many :grid_service_deploys, dependent: :destroy
+
+  def error!(message)
+    self.error_at = Time.now
+    self.error = message
+    self.save!
```

Comment:
Setters need `self.`, `save!` does not.

Reply:
I wonder if the model method should even do the `save!`

Reply:
It has the `!`so I guess yes. It could do `update(error: message, error_at: Time.now)` instead of the `=` + `save`.